### PR TITLE
Switched map default state to mapa patch instead of DOL patch

### PR DESCRIPTION
**Pull request by UltiNaruto**

### CODE REVIEW

File: structs/src/mapa.rs [hunk 1]
```diff
@@ -81,6 +84,21 @@ pub enum MapaObjectVisibilityMode
     MapStationOrVisit2 = 4,
 }
 
+impl FromStr for MapaObjectVisibilityMode {
+    type Err = ();
+
+    fn from_str(input: &str) -> Result<MapaObjectVisibilityMode, Self::Err> {
```
Comment (toasterparty) [inline structs/src/mapa.rs:100]: Please some method of mapping the original global enum values to preserve backwards compatibility. Per [Semantic Versioning](https://semver.org/), we would have to increment to randomprime v2.0 if we didn't preserve backwards compatibility here.

File: structs/src/mapa.rs [hunk 2]
```diff
@@ -81,6 +84,21 @@ pub enum MapaObjectVisibilityMode
...
+        match input {
```
Comment (toasterparty) [inline structs/src/mapa.rs]: `.strip()` input and make case insensitive please

**Review by toasterparty — CHANGES_REQUESTED** [review]

**Comment by UltiNaruto** [comment]
The global config for map default state is now the default for per room one if RoomConfig::map_default_state isn't referenced. If both aren't referenced then it defaults to MapStationOrVisit.

Also visited state is not a thing in MAPA file so if it's ever used use global map default state.

File: src/patch_config.rs [hunk 3]
```diff
@@ -505,6 +505,8 @@ pub struct PatchConfig
     pub heat_damage_per_sec: f32,
     pub staggered_suit_damage: bool,
     pub item_max_capacity: HashMap<PickupType, u32>,
+    #[deprecated = "Use RoomConfig::map_default_state instead of global map_default_state"]
```
Comment (toasterparty) [inline src/patch_config.rs]: I'd prefer the project to compile with 0 warnings, sorry. Just put a comment for now.

**Review by toasterparty — CHANGES_REQUESTED** [review]

File: src/patch_config.rs (hunk 3 above)
Comment (UltiNaruto) [reply src/patch_config.rs]: Done

**Pull request merged**